.github/actions/nuxeo/nos-publish/action.yml

```diff
@@ -0,0 +1,128 @@
+          export "PACKAGE_${i^^}=${VALUE}"
+          echo "PACKAGE_${i^^}=${VALUE}" | tee -a $GITHUB_ENV
+        done
+        if [[ -z "${PACKAGE_NAME}" || -z "${PACKAGE_VERSION}" ]]; then
+          echo "❌ Unable to extract package name or version from descriptor"
+          exit 1
+        fi
+
+    - name: Get Branch name
+      uses: Alfresco/alfresco-build-tools/.github/actions/get-branch-name@v10.1.0
+
+    - name: Evaluate Target env if none has been set
+      shell: bash
+      run: |
+        case "${{ inputs.nos-env }}" in
+          "production")
+            CONNECT_HOST=connect.nuxeocloud.com
+            ;;
+          "staging")
+            CONNECT_HOST=nos-preprod-connect.nuxeocloud.com
+            ;;
+          *)
+            if [[ "${BRANCH_NAME}" == "${{ github.event.repository.default_branch }}" ]]; then
+              CONNECT_HOST=connect.nuxeocloud.com
+            else
+              CONNECT_HOST=nos-preprod-connect.nuxeocloud.com
+            fi
+            ;;
+        esac
+        echo "CONNECT_HOST=${CONNECT_HOST}" >> $GITHUB_ENV
+        echo "🎯 Target Nuxeo Online Services environment: '${CONNECT_HOST}'"
+
+    - name: Publish package to Nuxeo Online Services
+      shell: bash
+      id: publish
+      env:
+        NOS_USERNAME: ${{ inputs.nos-username }}
+        NOS_TOKEN: ${{ inputs.nos-token }}
+      run: |
+        PUBLISH=$(curl -sX POST \
+          -u "${NOS_USERNAME}:${NOS_TOKEN}" \
+          -w "%{http_code}" \
+          -F "package=@${{ inputs.package-path }}" \
+          "https://${CONNECT_HOST}/nuxeo/site/marketplace/upload?batch=true&orgId=${{ inputs.nos-org }}")
+        echo "📢 Publishing package to Nuxeo Online Services..."
+        if [[ "${PUBLISH: -3}" != "200" ]]; then
+          echo "❌ Failed to publish package. HTTP Status: ${PUBLISH: -3}"
+          exit 1
+        fi
+
+    - name: Verify published package on Nuxeo Online Services
+      id: verify
+      env:
+        NOS_USERNAME: ${{ inputs.nos-username }}
+        NOS_TOKEN: ${{ inputs.nos-token }}
+      shell: bash
+      if: inputs.skip-verify == 'false'
+      run: |
+        sleep 5
+        echo "🔍 Verifying published package on Nuxeo Online Services..."
+        PUBLISHED=$(curl -sX GET \
+          -u "${NOS_USERNAME}:${NOS_TOKEN}" \
+          -w "%{http_code}" \
+          "https://${CONNECT_HOST}/nuxeo/site/marketplace/package/${PACKAGE_NAME}?version=${PACKAGE_VERSION}")
+        if [[ "${PUBLISHED: -3}" != "200" ]]; then
+          echo "❌ Published package verification failed. HTTP Status: ${PUBLISHED: -3}"
+          exit 1
+        fi
+        echo "✅ Package '${PACKAGE_NAME}' version '${PACKAGE_VERSION}' successfully published to Nuxeo Online Services."
+
+    - name: Populate outputs
+      id: publish-output
+      shell: bash
+      if: always()
+      env:
+        CONNECT_BASE_URL: https://${{ env.CONNECT_HOST }}/nuxeo/site/marketplace/package
```

Comment:
Could `/nuxeo/site/marketplace[/package]` be factorized? Or even with the Connect host?